File: src/NoiseSentinel.DAL/Models/Challan.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace NoiseSentinel.DAL.Models;

[Table("CHALLAN")]
public partial class Challan
{
    [Key]
    [Column("ChallanID")]
    public int ChallanId { get; set; }

    [Column("OfficerID")]
    public int? OfficerId { get; set; }

    [Column("AccusedID")]
    public int? AccusedId { get; set; }

    [Column("VehicleID")]
    public int? VehicleId { get; set; }

    [Column("ViolationID")]
    public int? ViolationId { get; set; }

    [Column("EmissionReportID")]
    public int? EmissionReportId { get; set; }

    [StringLength(255)]
    public string? EvidencePath { get; set; }

    [Column(TypeName = "datetime")]
    public DateTime? IssueDateTime { get; set; }

    [Column(TypeName = "datetime")]
    public DateTime? DueDateTime { get; set; }

    [StringLength(100)]
    public string? Status { get; set; }

    [StringLength(255)]
    public string? BankDetails { get; set; }

    [StringLength(255)]
    public string? DigitalSignatureValue { get; set; }

    [ForeignKey("AccusedId")]
    [InverseProperty("Challans")]
    public virtual Accused? Accused { get; set; }

    [ForeignKey("EmissionReportId")]
    [InverseProperty("Challans")]
    public virtual Emissionreport? EmissionReport { get; set; }

    [InverseProperty("Challan")]
    public virtual ICollection<Fir> Firs { get; set; } = new List<Fir>();

    [ForeignKey("OfficerId")]
    [InverseProperty("Challans")]
    public virtual Policeofficer? Officer { get; set; }

    [ForeignKey("VehicleId")]
    [InverseProperty("Challans")]
    public virtual Vehicle? Vehicle { get; set; }

    [ForeignKey("ViolationId")]
    [InverseProperty("Challans")]
    public virtual Violation? Violation { get; set; }
}

File: src/Noisesentinel.WebPortal/src/models/Challan.ts
// Challan DTOs (if not already exists)

export interface ChallanDto {
  challanId: number;

  // Officer Info
  officerId?: number;
  officerName?: string;
  badgeNumber?: string;
  officerBadgeNumber?: string;
  officerRank?: string;

  // Accused Info
  accusedId: number;
  accusedName?: string;
  accusedCnic?: string;
  accusedContact?: string;
  accusedAddress?: string;
  accusedCity?: string;
  accusedProvince?: string;
  accusedEmail?: string;

  // Vehicle Info
  vehicleId: number;
  plateNumber?: string;
  vehiclePlateNumber?: string; // Backend uses this property name
  vehicleMake?: string;
  vehicleColor?: string;
  vehicleMakeYear?: string;

  // Violation Info
  violationId: number;
  violationType?: string;
  penaltyAmount?: number;
  isCognizable?: boolean;

  // Emission Report (Evidence)
  emissionReportId?: number;
  deviceName?: string;
  soundLevelDBa?: number;
  mlClassification?: string;
  emissionTestDateTime?: string;

  // Challan Details
  evidencePath?: string;
  issueDateTime: string;
  dueDateTime: string;
  status: string;
  bankDetails?: string;
  digitalSignatureValue?: string;

  // Station Info
  stationId?: number;
  stationName?: string;

  // FIR Info
  hasFir: boolean;
  firId?: number;

  // Computed Fields
  issueDateFormatted?: string;
  dueDateFormatted?: string;
  daysUntilDue?: number;
  daysOverdue?: number;
  isOverdue?: boolean;
  integrityStatus?: string;
}

export interface ChallanSearchFilter {
  searchQuery?: string;
  stationIds?: number[];
  status?: string[];
  violationIds?: number[];
  isOverdue?: boolean;
  hasFir?: boolean;
  dateRange?: {
    start: string;
    end: string;
  };
  penaltyRange?: {
    min: number;
    max: number;
  };
}
